src/devices/device.ts

```diff
@@ -77,6 +77,14 @@ export class Device extends EventEmitter {
         return this._isWeDo2SmartHub;
     }
 
+    public writeDirect (mode: number, data: Buffer, callback?: () => void) {
+        if (this.isWeDo2SmartHub) {
+            this.send(Buffer.concat([Buffer.from([this.portId, 0x01, 0x02]), data]), Consts.BLECharacteristic.WEDO2_MOTOR_VALUE_WRITE);
+        } else {
+            this.send(Buffer.concat([Buffer.from([0x81, this.portId, 0x11, 0x51, mode]), data]), Consts.BLECharacteristic.LPF2_ALL, callback);
```

Comment:
Does it change the subscribed mode ?

Reply:
I'm not exactly sure. I don't think it changes subscribed mode _per se_. I think this changes the output mode of the device, ie. switching between accepting a triple RGB value vs a single color value, but doesn't actually subscribe to any notifications. But I have noticed in some scenarios it's necessary to send an explict subscribe first.

Reply:
I noticed two output modes on the color and distance : COL_O & IR_Tx.

I assume COL_O is some kind of color control for the led, I don't have one right now but we could  be able to understand what it does from its mode definition.

I think IR_Tx is some kind of serial message sending over IR, I don't have any use case for this but I find this idea somehow very exiting.

I wonder if having a mode definition is the difference between needing to subscribe before or not.